tests/test_full_model.py

```diff
@@ -53,138 +63,94 @@ def test_full_model_exact_match(tokenizer, use_peft, pass_empty_tensors, atol_fo
-            do_sample=True,
-            **sampling_options,
-        )
-    with torch.random.fork_rng():
-        hf_outputs_batch = HfGenerationMixin.sample(
-            model,
-            input_ids=inputs_batch,
-            max_length=inputs_batch.size(1) + max_new_tokens,
-            logits_warper=logits_warper,
-        )
-    assert torch.allclose(
-        remote_outputs_batch, hf_outputs_batch
-    ), "Sampling results are not identical to HF in multibatch mode"
+
+    for options in [
+        dict(do_sample=True),
+        dict(do_sample=True, temperature=0.5),
+        dict(do_sample=True, temperature=0.5, top_k=5),
+        dict(do_sample=True, temperature=0.5, top_k=5, top_p=0.9),
+        dict(do_sample=True, temperature=0.5, top_k=5, top_p=0.9, multiple_calls=True),
+        dict(do_sample=True, temperature=0.5, repetition_penalty=1.2),
+    ]:
+        options.update(max_new_tokens=max_new_tokens)
+        multiple_calls = options.pop("multiple_calls", False)
+
+        for inputs in [inputs_single, inputs_batch]:
+            torch.manual_seed(0)
+            outputs = make_generate_calls(model, inputs, multiple_calls=multiple_calls, **options)
+
+            torch.manual_seed(0)
+            ref_outputs = ref_model.generate(inputs, **options)
+
+            assert torch.allclose(
+                outputs, ref_outputs
+            ), f"Sampling is not identical to HF with {options=}, {inputs.shape=}"
 
 
 @pytest.mark.forked
-def test_beam_search_generation(tokenizer, max_new_tokens=4, num_beams=2):
-    model = AutoDistributedModelForCausalLM.from_pretrained(
-        MODEL_NAME, initial_peers=INITIAL_PEERS, torch_dtype=torch.float32
-    )
-    text = "A cat sat on a mat"
-    inputs = tokenizer(text, return_tensors="pt")["input_ids"]
-    remote_outputs = model.generate(
-        inputs,
-        max_new_tokens=max_new_tokens,
-        num_beams=num_beams,
-    )
-    beam_scorer = BeamSearchScorer(
-        batch_size=inputs.size(0),
-        num_beams=num_beams,
-        device=inputs.device,
-        length_penalty=0,
-        do_early_stopping=False,
-    )
-    hf_inputs = tokenizer([text] * 2, return_tensors="pt")["input_ids"]
-    hf_outputs = HfGenerationMixin.beam_search(
-        model, input_ids=hf_inputs, max_length=inputs.size(1) + max_new_tokens, beam_scorer=beam_scorer
-    )
-    assert torch.allclose(remote_outputs, hf_outputs), "Beam search results are not identical to HF"
+def test_beam_search_generation(tokenizer, model, ref_model, max_new_tokens=4, num_beams=5):
+    inputs = tokenizer("A cat sat on a mat", return_tensors="pt")["input_ids"]
+
+    options = dict(max_new_tokens=max_new_tokens, num_beams=num_beams, do_sample=False)
+    outputs = make_generate_calls(model, inputs, **options)
+    ref_outputs = ref_model.generate(inputs, **options)
+    assert torch.allclose(outputs, ref_outputs), f"Beam search results are not identical to HF with {multiple_calls=}"
```

Comment:
multple_calls is not defined. Please set up a parameter (multiple_calls) via pytest.parameterize to test that we 
(at least) do not raise an error here

Reply:
I've just fixed the undefined var for now. I've manually checked that the warning is shown but don't want to commit to this behavior in tests.